src/UI/UIUtils.java

```diff
@@ -157,13 +158,25 @@ else if(fileEntry.canRead() && !fileEntry.getName().endsWith(DEFAULT_FACTION_NAM
   public static ColorPalette getBuildingColors(Color colorKey)
   {
     initCosmetics();
-    return buildingColorPalettes.get(colorKey);
+    ColorPalette palette = buildingColorPalettes.get(colorKey);
+    if (null == palette) // Uh oh, the player's messing with us. Make stuff up so we don't crash.
+    {
+      buildingColorPalettes.put(colorKey, buildingColorPalettes.get(Color.PINK));
```

Comment:
At least print a warning here, and also in `getMapUnitColors`.

Reply:
Fixed in `moldyInfo`